src/providers/local/exit-pool.provider.ts

```diff
@@ -0,0 +1,195 @@
+import { isDeep, tokenTreeNodes } from '@/composables/usePool';
+import useTokens from '@/composables/useTokens';
+import { useTxState } from '@/composables/useTxState';
+import {
+  HIGH_PRICE_IMPACT,
+  REKT_PRICE_IMPACT,
+} from '@/constants/poolLiquidity';
+import symbolKeys from '@/constants/symbol.keys';
+import { fetchPoolsForSor, hasFetchedPoolsForSor } from '@/lib/balancer.sdk';
+import { bnum, removeAddress, trackLoading } from '@/lib/utils';
+import { Pool } from '@/services/pool/types';
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { debounce } from 'lodash';
+import {
+  computed,
+  defineComponent,
+  h,
+  InjectionKey,
+  onBeforeMount,
+  onBeforeUnmount,
+  PropType,
+  provide,
+  ref,
+  toRef,
+  watch,
+} from 'vue';
+
+/**
+ * TYPES
+ */
+type Props = {
+  pool: Pool;
+};
+
+/**
+ * ExitPoolProvider
+ *
+ * Handles pool exiting state and transaction execution.
+ */
+const provider = (props: Props) => {
+  /**
+   * STATE
+   */
+  const pool = toRef(props, 'pool');
```

Comment:
Could this be a computed property so it stays in sync with the `pool` prop?

Reply:
This is how toRef works: https://vuejs.org/api/reactivity-utilities.html#toref

"The created ref is synced with its source property: mutating the source property will update the ref, and vice-versa."

But either way is fine.

Reply:
Ah, I somehow missed that. All good then 👍